internal/planner/planner.go

```diff
@@ -992,23 +995,28 @@ func (p *Planner) planExprCallFunc(name string, arity int, void bool, operands [
 
 			return iter()
 		})
-	}
 
-	// definition: f(x) = y { ... }
-	// call: f(x, 1) # caller captures result
-	return p.planCallArgs(operands[:len(operands)-1], 0, args, func(args []ir.Operand) error {
-		result := p.newLocal()
-		p.appendStmt(&ir.CallStmt{
-			Func:   name,
-			Args:   args,
-			Result: result,
+	case len(operands) == arity+1:
+		// definition: f(x) = y { ... }
+		// call: f(x, 1) # caller captures result
+		return p.planCallArgs(operands[:len(operands)-1], 0, args, func(args []ir.Operand) error {
+			result := p.newLocal()
+			p.appendStmt(&ir.CallStmt{
+				Func:   name,
+				Args:   args,
+				Result: result,
+			})
+			return p.planUnifyLocal(op(result), operands[len(operands)-1], iter)
 		})
-		return p.planUnifyLocal(op(result), operands[len(operands)-1], iter)
-	})
+
+	default:
```

Comment:
I've tightened the checks here because they've been quite loose. This should never happen, but now it's an error.